### ISSUE TITLE
Types: zustand v4

### ISSUE THREAD

**Issue by hirbod** [cmdominguez/react-native-flipper-zustand#3]
Please add support for zustand v4 types.

https://github.com/pmndrs/zustand/blob/main/docs/typescript.md
https://github.com/pmndrs/zustand/blob/main/docs/v4-migration.md

**Comment by AronBe** [comment]
Yeah I cannot figure it out either, @cmdominguez would you have a possibility to look into it?

**Comment by AronBe** [comment]
@cmdominguez - thanks for taking time to work on this 👍 

the typing for string as a third param looks ok if used on its own, but the store type itself seems to be `never`, also wondering how to make the types work with another frequent middleware such as `persist`

![image](https://user-images.githubusercontent.com/32416348/173516731-aaf539df-0bc1-407a-af62-7534948a89bc.png)

**Comment by rpadilla6** [comment]
@hirbod I made a patch for this based on zustand documentation for typing middleware, the current typings were almost there but I think were trying to do too much. I'll link the issue here in a sec

**Comment by WarrenBuffering** [comment]
+1, also looking for an update, or even an alternative would be fine at this point
